crates/rethnet_eth/src/remote/cacheable_method_invocation.rs

```diff
@@ -0,0 +1,154 @@
+use crate::remote::methods::{GetLogsInput, MethodInvocation, Percentiles, TransactionInput};
+use crate::remote::BlockSpec;
+use crate::U256;
+use revm_primitives::{Address, B256};
+
+/// These method invocations are hashable and can be potentially cached.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+// Using repr(u8) to let us assign a number to each variant which is important to make sure hashing
+// remains consistent if new variants are added or if variants are reordered.
+#[repr(u8)]
+pub(super) enum CacheableMethodInvocation<'a> {
+    /// eth_call
+    Call(&'a TransactionInput, &'a Option<BlockSpec>) = 0,
```

Comment:
`BlockSpec` is not deterministic. When it is used, the "earliest", "safe", "latest", etc. tags need to be resolved based on the received response.

I'd recommend using an `Option<U256>` for the block number here instead. Also, see my comment on the previous PR about resolvers

Reply:
Thanks, I'm planning to address this in https://github.com/NomicFoundation/rethnet/issues/88, so I'd say it's out of scope now unless you want me to combine the two PRs into one.